SetupDataPkg/Plugins/GenSetupDataBin/GenSetupDataBin.py

```diff
@@ -123,41 +96,98 @@ def do_pre_build(self, thebuilder):
 
     # Attempt to run GenCfgData to generate setup data binary blob, output will be placed at
-    # "MSFT_PLATFORM_PACKAGE"/BuiltInVars.xml
+    # ConfPolicyVarBin_*.bin
+    #
+    # Consumes build environment variables: "BUILD_OUTPUT_BASE", "YAML_CONF_FILE", "XML_CONF_FILE",
+    # "DELTA_CONF_POLICY" (optional), "CSV_CONF_POLICY" (optional)
+    def do_pre_build(self, thebuilder):
+        # Generate Generic Profile
+        self.generate_profile(thebuilder, None, None, 0)
+
+        # Build other profiles, if present
+        delta_conf = thebuilder.env.GetValue("DELTA_CONF_POLICY")
+        csv_conf = thebuilder.env.GetValue("CSV_CONF_POLICY")
+
+        if (delta_conf is None and csv_conf is not None and thebuilder.env.GetValue("YAML_CONF_FILE") is not None) or\
+                (delta_conf is not None and csv_conf is None and thebuilder.env.GetValue("XML_CONF_FILE") is not None):
+            logging.error("There must be an equal number of entries for delta and csv files when both YAML/XML present")
+            return -1
+
+        if delta_conf is not None and csv_conf is not None:
+            delta_conf = delta_conf.split(";")
+            csv_conf = csv_conf.split(";")
+            # Validate we have the same number of delta and csv files, as required
+            if len(delta_conf) != len(csv_conf):
+                logging.error("Differing number of Delta and CSV files provided for profiles, they must be the same.")
+                return -1
+
+            for idx in range(len(delta_conf)):
+                # Validate the name of the delta file and csv file match, to mitigate human error in placement
+                # ignore file suffix
+                if delta_conf[idx][:-4] != csv_conf[idx][:-4]:
+                    logging.error("Delta and CSV files do not have the same name, list might be in the wrong order.")
+                    return -1
+
+                # Generate the profile
+                self.generate_profile(thebuilder, delta_conf[idx], csv_conf[idx], idx + 1)
+        elif delta_conf is not None:
+            delta_conf = delta_conf.split(";")
+
+            for idx in range(len(delta_conf)):
+                # Generate the profile
+                self.generate_profile(thebuilder, delta_conf[idx], None, idx + 1)
+        elif csv_conf is not None:
+            csv_conf = csv_conf.split(";")
+
+            for idx in range(len(csv_conf)):
+                # Generate the profile
+                self.generate_profile(thebuilder, None, csv_conf[idx], idx + 1)
+        else:
+            logging.warn("Either DELTA_CONF_POLICY or CSV_CONF_POLICY or both not set. Only generic profile generated.")
+
+        return 0
+
+    # Delete intermediate binary files
     #
-    # Consumes build environment variables: "MSFT_PLATFORM_PACKAGE"
+    # Consumes build environment variables: "DELTA_CONF_POLICY" and "CSV_CONF_POLICY"
     def do_post_build(self, thebuilder):

```

Comment:
Just curious, do you really want to remove these files after a build? Would someone want to check it?

Reply:
I'm okay to leave them, I was just cleaning up extra bins that won't be used, as they were just intermediate bins that get combined into the profile bins.